api/client_test.go

```diff
@@ -409,63 +410,112 @@ func TestClientNonTransportRoundTripper(t *testing.T) {
 }
 
 func TestClone(t *testing.T) {
-	client1, err := NewClient(DefaultConfig())
-	if err != nil {
-		t.Fatalf("NewClient failed: %v", err)
+	type fields struct {
+		config  *Config
+		headers *http.Header
 	}
-
-	// Set all of the things that we provide setter methods for, which modify config values
-	err = client1.SetAddress("http://example.com:8080")
-	if err != nil {
-		t.Fatalf("SetAddress failed: %v", err)
+	tests := []struct {
+		name   string
+		fields fields
```

Comment:
Is the intermediate `fields` needed or can we have *Config and *http.Header as part of the struct field directly?

Reply:
Absolutely.